main.py

```diff
@@ -66,42 +69,56 @@ def calculate_unadaptability_metrics(
 
     return ufm_pre_acc, ufm_fine_acc, ufm_fine_losses
 
-# For recording tracebacks on CAIS:
-# https://cluster.safe.ai/#enabling-debugging-for-distributed-training
-@record
-def main():
-    # Parse args and the config file
-    args, config = cli.parse_arguments()
-
+def init_run(cfg: DictConfig) -> None:
+    """
+    Initialize the run
+    You can later access the Run object with wandb.run
+    """
     wandb.init(
-        project="unadaptable-foundation-models", config=dict(config),
+        project=cfg.get("project", "unadaptable-foundation-models"),
+        # Convert config to dict type
+        config=OmegaConf.to_container(cfg, resolve=True),
         mode="disabled" if args.disable_wandb else "online",
-        dir=config["store_path"]
+        tags=cfg.get("tags", None),
+        # dir=cfg["store_path"],
     )
 
-    seed_all(config.seed)
+    seed_all(cfg.seed)
     device = "cuda" if torch.cuda.is_available() else "cpu"
     print(f"Using {device} device")
 
-    model = get_model(config.model, device)
+    return
+
+# For recording tracebacks on CAIS:
+# https://cluster.safe.ai/#enabling-debugging-for-distributed-training
+@record
```

Comment:
We prob won't need distributed training. Would suggest leaving this out until we prove that we need it.